src/server/pfs/server/driver_v2.go

```diff
@@ -513,3 +515,55 @@ func (d *driverV2) compactionWorker() {
 	// Never ending backoff should prevent us from getting here.
 	panic(err)
 }
+
+func (d *driverV2) copyFile(pachClient *client.APIClient, src *pfs.File, dst *pfs.File, overwrite bool) (retErr error) {
+	if overwrite {
+		// TODO: after delete merging is sorted out add overwrite support
+		return errors.New("overwrite not yet supported")
+	}
+	return d.withFileSet(pachClient.Ctx(), dst.Commit.Repo.Name, dst.Commit.ID, func(dstFs *fileset.FileSet) error {
+		return d.getTarConditional(pachClient.Ctx(), src.Commit.Repo.Name, src.Commit.ID, src.Path, func(rf *FileReader) error {
```

Comment:
This is not the way we want to do this. This does not take advantage of the cheap copying. This should be doing essentially what the storage compact function does (setup a merge reader from the compaction levels and writing it to an output fileset). I think we might want support for filtering based on globs, but a simple single file copy is probably fine for the time being. So, this could just make a compact call on the src compaction levels with the index option being the src file path. I think you might hit some trickiness with handling the renaming since the content stream has file metadata (the tar header). This will probably require a bit more thought, but I think the basic approach would be to read the tar header, apply the change, then cheap copy the content.